tests/ext/test_filters.py

```diff
@@ -1877,6 +1878,13 @@ def test_filters_successful_payment(self, update):
         update.message.successful_payment = "test"
         assert filters.SUCCESSFUL_PAYMENT.check_update(update)
 
+    def test_filters_successful_payment_payloads(self, update):
+        update.message.successful_payment = SuccessfulPayment(
+            "USD", 100, "custom-payload", "123", "123"
+        )
+        assert filters.SuccessfulPayment(("custom-payload",)).check_update(update)
+        assert not filters.SuccessfulPayment(["test1"]).check_update(update)
```

Comment:
please also add a test checking the string representation of the new class. There are a few tests with "repr" in their name in this file that you can look at for inspiration :)
It would also be nice

* if you can add a line with `SuccesfulPayment().check_update` here for completenes so that this is not only implicitly tested via `SUCCESFUL_PAYMENT`
* you can test with a message taht doesn't contain a successful payment at all

Reply:
> add a test checking the string representation of the new class.

Is it okay to keep the str repr of the filter, as it for e.g will return, `filters.SuccessfulPayment(['payload'])`, or should it be modified to match some other filters that implement a `name` property to remove the square brackets and single quotes, so like, `filters.SuccessfulPayment(payload1, payload2)`

Reply:
Sounds fine to me. I haven't found other filters that do what you describe, tbh :D

Reply:
I agree it's not necessary. However the filter I was referring to was `filters._ChatUserBaseFilter`.
https://github.com/python-telegram-bot/python-telegram-bot/blob/2345bfbb5337f6e5d63b9d56bbf3fa75646c1b35/telegram/ext/filters.py#L802-L807